api/applications/serializers.py

```diff
@@ -99,8 +102,26 @@ def create(self, validated_data):
             raise exceptions.ValidationError(detail=detail)
         return instance
 
+
+class ApiOAuth2ApplicationDetailSerializer(ApiOAuth2ApplicationSerializer):
+    """
+    Overrides ApiOAuth2ApplicationSerializer to make id required
+    and client_secret writable to reset the secret via API patch request
+    """
+
+    id = IDField(source='client_id', required=True, help_text='The client ID for this application (automatically generated)')
+
+    client_secret = ser.CharField(
+        help_text='The client secret for this application (automatically generated)',
+        allow_null=True,
+        required=False,
+    )
+
     def update(self, instance, validated_data):
         assert isinstance(instance, ApiOAuth2Application), 'instance must be an ApiOAuth2Application'
+        client_secret = validated_data.pop('client_secret', None)
+        if client_secret == '':
+            instance.reset_secret(save=True)
```

Comment:
To reset the secret, request body will now look like this:
```json
{
	"data": {
		"id": "application_idfkdsljfjasd",
		"type": "applications",
		"attributes": {
		   "client_secret": null
		}
	}
}
```
with allow_null is true, `null` will be converted to an empty string after data is validated.  allow_blank is not true, so an empty string can't be passed in into the request